systems/subscriber/registry/pkg/db/subscriber_repo_test.go

```diff
@@ -187,101 +527,451 @@ func TestSubscriber_Get(t *testing.T) {
-			AddRow(networkId)
+		networkId := uuid.NewV4()
+		subscriber := createTestSubscriber("Alice", "alice@example.com")
+		subscriber.NetworkId = networkId
+
+		rows := sqlmock.NewRows([]string{
+			"subscriber_id", "name", "network_id", "email", "phone_number",
+			"gender", "dob", "proof_of_identification", "id_serial", "address",
+			"created_at", "updated_at", "deleted_at",
+		}).AddRow(
+			subscriber.SubscriberId, subscriber.Name, subscriber.NetworkId,
+			subscriber.Email, subscriber.PhoneNumber, subscriber.Gender,
+			subscriber.DOB, subscriber.ProofOfIdentification, subscriber.IdSerial,
+			subscriber.Address, subscriber.CreatedAt, subscriber.UpdatedAt, subscriber.DeletedAt,
+		)
+
+		mock.ExpectQuery(`^SELECT \* FROM "subscribers" WHERE "subscribers"."network_id" = \$1`).
+			WithArgs(networkId).
+			WillReturnRows(rows)
 
-		mock.ExpectQuery(`^SELECT.*subscribers.*`).
-			WithArgs(networkId, sqlmock.AnyArg()).
-			WillReturnRows(subRow)
+		// Act
+		subscribers, err := repo.GetByNetwork(networkId)
 
+		// Assert
 		assert.NoError(t, err)
-		repo := int_db.NewSubscriberRepo(&UkamaDbMock{
-			GormDb: gdb,
+		assert.Len(t, subscribers, 1)
+		assert.Equal(t, subscriber.SubscriberId, subscribers[0].SubscriberId)
+		assert.Equal(t, subscriber.Name, subscribers[0].Name)
+		assert.Equal(t, subscriber.Email, subscribers[0].Email)
+		assert.Equal(t, networkId, subscribers[0].NetworkId)
+		assert.NoError(t, mock.ExpectationsWereMet())
+	})
+
+	t.Run("SuccessWithNoSubscribers", func(t *testing.T) {
+		// Arrange
+		db, mock, repo := setupTestDB(t)
+		defer db.Close()
+
+		networkId := uuid.NewV4()
+
+		rows := sqlmock.NewRows([]string{
+			"subscriber_id", "name", "network_id", "email", "phone_number",
+			"gender", "dob", "proof_of_identification", "id_serial", "address",
+			"created_at", "updated_at", "deleted_at",
 		})
+
+		mock.ExpectQuery(`^SELECT \* FROM "subscribers" WHERE "subscribers"."network_id" = \$1`).
+			WithArgs(networkId).
+			WillReturnRows(rows)
+
 		// Act
-		sub, err := repo.Get(networkId)
+		subscribers, err := repo.GetByNetwork(networkId)
 
 		// Assert
 		assert.NoError(t, err)
+		assert.Len(t, subscribers, 0)
+		assert.NoError(t, mock.ExpectationsWereMet())
+	})
 
-		err = mock.ExpectationsWereMet()
+	t.Run("DatabaseError", func(t *testing.T) {
+		// Arrange
+		db, mock, repo := setupTestDB(t)
+		defer db.Close()
+
+		networkId := uuid.NewV4()
+		dbErr := errors.New("database connection error")
+
+		mock.ExpectQuery(`^SELECT \* FROM "subscribers" WHERE "subscribers"."network_id" = \$1`).
+			WithArgs(networkId).
+			WillReturnError(dbErr)
+
+		// Act
+		subscribers, err := repo.GetByNetwork(networkId)
+
+		// Assert
+		assert.Error(t, err)
+		assert.Equal(t, dbErr, err)
+		assert.Nil(t, subscribers)
+		assert.NoError(t, mock.ExpectationsWereMet())
+	})
+}
+
+func TestSubscriber_Delete(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		db, mock, repo := setupTestDB(t)
```

Comment:
we need to hide the sqlmock.DB. We only need the `repo` and the `mock`

Reply:
addressed